File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n=nums.length;int count=0,left=0,sum=0,ml=Integer.MAX_VALUE;

        for(int i=0;i<n;i++){

             sum +=nums[i];
           count++;
          

           while(sum>=target) {
               sum -=nums[left];
               left++;
               
            ml=Math.min(ml,count);
               count--;
           }

          
        }
       return ml == Integer.MAX_VALUE ? 0 : ml; // If no subarray found, return 0
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        s = 0 
        e = 0 
        n = len(nums)
        minlen = float('inf')
        sum = 0

        while e < n:
            sum += nums[e]
            
            while sum >= target:
                minlen =  min(minlen , e - s + 1)
                sum -= nums[s]
                s += 1

            e += 1    

        return 0 if minlen == float('inf') else minlen


